### PR TITLE
MBS-8867: Don't uppercase "C'mon"

### PR DESCRIPTION
**Pull request by reosarevok**
https://tickets.metabrainz.org/browse/MBS-8867

All eslint fixes except for the first commit (+7|-1) and the last one (which just fixes a couple typos etc in comment strings)

### CODE REVIEW

File: root/static/scripts/guess-case/utils.js [hunk 1]
```diff
@@ -173,7 +173,8 @@ export function titleString(gc, is, forceCaps) {
     // we got an 'til (Until = 'til), lowercase.
     // we got an 'way (Away = 'way), lowercase.
     // we got an 'round (Around = 'round), lowercased
-    } else if (isApostrophe(gc.i.getPreviousWord()) && lc.match(/^(s|round|em|ve|ll|d|cha|re|til|way|all)$/i)) {
+    // we got a 'mon (Come on = C'mon), lowercase
```
Comment (mwiencek) [inline root/static/scripts/guess-case/utils.js]: Should we uppercase the C below, too (in the "Make it O'Titled, Y'All" part)? (If so an extra small test for that would be nice.)

Comment (reosarevok) [reply root/static/scripts/guess-case/utils.js]: Good idea. Just changed the new test a bit for both.

**Comment by mwiencek** [comment]
I think the "Regex improvements" commit somehow got squashed with a bunch of eslint changes. It'd be nice to split those because I can't see what any of the actual regex changes are.

**Comment by reosarevok** [comment]
ARGH. Will fix.

**Comment by reosarevok** [comment]
Split that. There were two different eslint commits there too, but that's harder to split so hopefully it's ok.

**Comment by mwiencek** [comment]
Much better now. :) Though there's one other small issue: 44121aeb6d3376cf265f0cdc13cb83c6789f67e9 changes the Make it O'Titled, Y'All regexp from `/^(c|o|y)$/i` to `/^(coy)$/i` which is broken and not an eslint change. It'd be good to revert that so that the intermediary commits don't introduce any unexpected breaking changes (which can be a PITA when using git bisect).

**Comment by reosarevok** [comment]
Good catch! Fixed :)

**Pull request merged**